### PR TITLE
Feature/docker

### PR DESCRIPTION
**Pull request by ilstefano**
PR to collect all commits concerning docker integartion

### CODE REVIEW

File: .docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,42 @@
+FROM php:8-fpm-alpine as runtime
+
+RUN apk update \
+  && apk add --no-cache curl make coreutils icu-libs icu-dev libzip-dev libxslt-dev php-mysqli php-pdo_mysql postgresql-dev libpng-dev
+
+RUN NPROC=$(grep -c ^processor /proc/cpuinfo 2>/dev/null || 1) \
+    && docker-php-ext-install -j${NPROC} zip intl xsl mysqli pdo_mysql pdo_pgsql gd
+
+FROM runtime AS build
```
Comment (Chemaclass) [inline .docker/Dockerfile]: The first FROM is clearly mandatory, but are these "extra FROM" statements necessary?
```Dockerfile
FROM runtime AS build
...
FROM build AS dev
...
```

File: README.md [hunk 2]
```diff
@@ -36,3 +36,116 @@ application-name
...
+
+#### Preliminary
+
+If you do local development with a [typical dockerized environment preconfigured](https://docs.docker.com/get-docker/) you can docker-build and use a docker image as runtime. All docker-build-information is in ```./.docker```.
+
+So we docker-build the image once:
+
+```bash
+# docker-build runtime image once
+docker build -t gacela \
```
Comment (Chemaclass) [inline README.md]: What about changing the container name from gacela to gacela-example. In this case it will be more explicit what is behind the container... Or, maybe for this gacela would be ok. What do you think, @JesusValera @ilstefano?

**Review by Chemaclass — APPROVED** [review]
Pretty cool idea! 💡 I just got some minor observations.

File: .docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,42 @@
+FROM php:8-fpm-alpine as runtime
+
+RUN apk update \
+  && apk add --no-cache curl make coreutils icu-libs icu-dev libzip-dev libxslt-dev php-mysqli php-pdo_mysql postgresql-dev libpng-dev
```
Comment (JesusValeraDev) [inline .docker/Dockerfile]: @mmscondo thank you so much for the PR.

Just one mention, I see you are installing some dependencies that are not required at all for this project like `make`, `libpng-dev` or even `mysql` and `postgresql`.

I would remove them, what do you think?

File: README.md (hunk 2 above)
Comment (JesusValeraDev) [reply README.md]: Yeah, I think `gacela-example` fits better, as we have already the `gacela` repository :)

File: .docker/Dockerfile (hunk 1 above)
Comment (Chemaclass) [reply .docker/Dockerfile]: Answer: https://github.com/ilstefano/gacela-example/pull/1#commitcomment-73399516
> Not yet. These are instructions to separate build commands into 'stages' (FROM <stage>) which help to easily use the same Dockerfile and `docker build' command during a development lifecycle. As IMHO everybody sooner or later builds her images for integrations or deployments to other docker environments (test, prod) this makes sense to leave it already here.
> TL;DR: are not needed technically without a complexer workflow, but spend semantical added value

Comment (Chemaclass) [reply .docker/Dockerfile]: Thanks for clarifying. I am totally ok with that :)

**Comment by Chemaclass** [comment]
Is the same person behind @mmscondo and @ilstefano? 😄

**Comment by Chemaclass** [comment]
Btw, this PR is pointing to your master branch, instead the origin repo `gacela-project/gacela-example@master`. If you change the pointing repo, we can merge it in the gacela-project 😉

**Comment by ilstephano** [comment]
Yes … to be honest, I’m also flashed and I don’t why these avatars get mixed up at GitHub …

Von: Jose Maria Valera Reales ***@***.***>
Gesendet: Mittwoch, 11. Mai 2022 15:28
An: ilstefano/gacela-example ***@***.***>
Cc: Lo, Stephan ***@***.***>; Mention ***@***.***>
Betreff: Re: [ilstefano/gacela-example] Feature/docker (PR #1)


Is the same person behind @mmscondo<https://github.com/mmscondo> and @ilstefano<https://github.com/ilstefano>? 😄

**Comment by ilstephano** [comment]
Thx … I just wanted to have some sort of ‘hidden’  PR without intimidating your repo…

Von: Jose Maria Valera Reales ***@***.***>
Gesendet: Mittwoch, 11. Mai 2022 15:30
An: ilstefano/gacela-example ***@***.***>
Cc: Lo, Stephan ***@***.***>; Mention ***@***.***>
Betreff: Re: [ilstefano/gacela-example] Feature/docker (PR #1)


Btw, this PR is pointing to your master branch, instead the origin repo ***@***.***

**Comment by Chemaclass** [comment]
This is the magic of OSS, no need to create "hidden PRs". Especially creating such awesome contribution! 🚀

File: .docker/Dockerfile (hunk 3 above)
Comment (ilstefano) [reply .docker/Dockerfile]: Yes I removed the extensions. I'll argue in the next PR why they existed.

**Comment by Chemaclass** [comment]
I wonder why you closed the PR. Are you creating another PR with those changes? You can create little PRs as well and work on little iterations as well. As you wish :)

**Comment by Chemaclass** [comment]
@ilstefano, check out this doc to fix your git username: 
https://docs.github.com/en/get-started/getting-started-with-git/setting-your-username-in-git 
```bash
git config --global user.name "ilstefano"
git config --global user.email "[your github email]"
```

**Comment by ilstefano** [comment]
Thanx for the git config hint! I guess I fixed this issue.

Meanwhile I closed this PR and continued it here: https://github.com/gacela-project/gacela-example/pull/5

I apologize any inconveniences. But I guess I need better used to github - local rebasing and reconciling this to upstreams didn't work ootb, thanx very much for your patience!
